independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -650,6 +650,20 @@ static void validateInterceptorDecorator(BeanInfo bean, List<Throwable> errors,
 
     static void validateBean(BeanInfo bean, List<Throwable> errors, Consumer<BytecodeTransformer> bytecodeTransformerConsumer,
             Set<DotName> classesReceivingNoArgsCtor) {
+        // validate injection points, this includes producer method params, disposer method params and so on
+        List<InjectionPointInfo> injectionPoints = new ArrayList<>();
+        injectionPoints.addAll(bean.getAllInjectionPoints());
+        if (bean.getDisposer() != null) {
+            injectionPoints.addAll(bean.getDisposer().getAllInjectionPoints());
+        }
+        for (InjectionPointInfo ipi : injectionPoints) {
```

Comment:
**Note to self:** In case of class beans, we are partly duplicating the iteration for the sake of IP `private` modifier transformation - see line 730 below. I am not exactly happy about that; we could extract that into a single for cycle and do it all in one go.

Reply:
This seems fine to me. I imagine merging the two loops would probably lead to harder-to-read code.